api/Dockerfile

```diff
@@ -19,44 +29,50 @@ RUN apt-get update --fix-missing && apt-get -y install r-base
 # Install cffdrs
 RUN R -e "install.packages('cffdrs')"
 
+# Install some other dependancies
+RUN apt-get -y install git build-essential python3 python3-dev python3-pip curl vim
+
 # Install JDK
 RUN apt-get -y install openjdk-11-jdk
 
-# Install pdf-kit for PDF generation on HFI (with qt support!)
-# As of writing, 0.12.6 (the most recent) is the version currently being installed by 3.8.12-buster,
-# but without qt. We do a manual instllation of 0.12.6 in order to get qt support.
-RUN apt-get update --fix-missing && apt-get -y install xfonts-75dpi xfonts-base
-RUN wget https://github.com/wkhtmltopdf/packaging/releases/download/0.12.6-1/wkhtmltox_0.12.6-1.buster_amd64.deb -P /tmp && \
-    dpkg -i /tmp/wkhtmltox_0.12.6-1.buster_amd64.deb
+# We could install poetry manually, but it's easier to use apt.
+RUN apt-get -y install python3-poetry
+# Poetry expects "python", but by default, on ubuntu, you need to specify "python3", so
+# we work around that, by using the python3-poetry command.
+RUN apt-get -y install python-is-python3
 
-# Update pip
-RUN python -m pip install --upgrade pip
+# from: https://code.visualstudio.com/remote/advancedcontainers/add-nonroot-user
+RUN groupadd --gid $USER_GID $USERNAME \
+    && useradd --uid $USER_UID --gid $USER_GID -m $USERNAME
 
-# Install gdal
-# We don't have much control over what version of gdal we're getting, it's pretty much whatever is
-# available to us. As such, gdal is not installed by poetry, since the version will differ between
-# platforms.
-RUN python -m pip install gdal==$(gdal-config --version)
+RUN mkdir /app
+RUN chown worker /app
+USER worker
+ENV PATH="/home/worker/.local/bin:${PATH}"
+
+WORKDIR /app
 
-# Install poetry
-RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | POETRY_HOME=/opt/poetry python && \
-    cd /usr/local/bin && \
-    ln -s /opt/poetry/bin/poetry && \
-    poetry config virtualenvs.create false
+# Update pip
+RUN python3 -m pip install --upgrade pip
 
 # Copy poetry files.
-COPY pyproject.toml poetry.lock /tmp/
+COPY pyproject.toml poetry.lock ./
 
-# Install dependancies.
-RUN cd /tmp && \
-    poetry install --no-root
+COPY --chown=worker:worker poetry.lock pyproject.toml ./
+RUN python3 -m pip install cachecontrol
+
+RUN poetry install
+
+# We can't have this inside pyproject.toml because the gdal version differs from platform to platform.
+# To figure out what version of pygdal you need, run gdal-config
+RUN poetry run python -m pip install pygdal==3.4.1.10
```

Comment:
Does the latest `pygdal` (3.5.1.11) work?
https://pypi.org/project/pygdal/3.5.1.11/

Reply:
No. Latest version that comes with Ubuntu 22.04 is 3.4.1